src/passthrough/sync_io.rs

```diff
@@ -36,6 +36,26 @@ impl<S: BitmapSlice + Send + Sync> PassthroughFs<S> {
         Self::open_proc_file(&self.proc_self_fd, file.as_raw_fd(), new_flags, data.mode)
     }
 
+    /// Check the HandleData flags against the flags from the current request
+    /// if these do not match, check the current flags with fcntl F_GETFL and
+    /// finally set the flags with F_SETFL if these again do not match
+    fn check_fd_flags(&self, data: Arc<HandleData>, fd: RawFd, flags: u32) -> io::Result<()> {
+        let open_flags = data.get_flags();
+        if open_flags != flags {
+            let curr_flags = unsafe { libc::fcntl(fd, libc::F_GETFL) };
+            if curr_flags < 0 {
+                return Err(io::Error::last_os_error());
+            }
+            if flags != curr_flags as u32 {
+                let ret = unsafe { libc::fcntl(fd, libc::F_SETFL, flags) };
```

Comment:
Should we update the `open_flag` stored in `HandleData` after changing the flags on host by `fcntl`?
Otherwise we may run into an inconsistent state if guest change the flags multiple times.

Reply:
Or it will cause redundant call to`fcntl(G_GETFL)`.

Reply:
Yes I think this makes sense, I have pushed a new commit wrapping HandleData with RwLock